pkg/kudoctl/cmd/update.go

```diff
@@ -0,0 +1,109 @@
+		kubectl kudo upgrade dev-flink -n services -p param=value`
+)
+
+type updateOptions struct {
+	Namespace  string
+	Parameters map[string]string
+}
+
+// defaultOptions initializes the install command options to its defaults
+var defaultUpdateOptions = &updateOptions{
+	Namespace: "default",
+}
+
+// newUpgradeCmd creates the install command for the CLI
+func newUpdateCmd() *cobra.Command {
+	options := defaultUpdateOptions
+	var parameters []string
+	upgradeCmd := &cobra.Command{
+		Use:     "update <instance-name>",
+		Short:   "Update installed KUDO operator.",
+		Long:    `Update installed KUDO operator with new parameters.`,
+		Example: updateExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Prior to command execution we parse and validate passed parameters
+			var err error
+			options.Parameters, err = install.GetParameterMap(parameters)
+			if err != nil {
+				return errors.WithMessage(err, "could not parse parameters")
+			}
+			return runUpdate(args, options)
+		},
+		SilenceUsage: true,
+	}
+
+	upgradeCmd.Flags().StringArrayVarP(&parameters, "parameter", "p", nil, "The parameter name and value separated by '='")
+	upgradeCmd.Flags().StringVar(&options.Namespace, "namespace", defaultOptions.Namespace, "The namespace where the instance you want to upgrade is installed in.")
+
+	const usageFmt = "Usage:\n  %s\n\nFlags:\n%s"
+	upgradeCmd.SetUsageFunc(func(cmd *cobra.Command) error {
+		fmt.Fprintf(upgradeCmd.OutOrStderr(), usageFmt, upgradeCmd.UseLine(), upgradeCmd.Flags().FlagUsages())
+		return nil
+	})
+	return upgradeCmd
+}
+
+func validateUpdateCmd(args []string, options *updateOptions) error {
+	if len(args) != 1 {
+		return errors.New("expecting exactly one argument - name of the instance installed in your cluster")
+	}
+	if len(options.Parameters) == 0 {
+		return errors.New("Need to specify at least one parameter to override via -p otherwise there is nothing to update")
+	}
+
+	return nil
+}
+
+func runUpdate(args []string, options *updateOptions) error {
+	err := validateUpdateCmd(args, options)
+	if err != nil {
+		return err
+	}
+	instanceToUpdate := args[0]
+
+	kc, err := kudo.NewClient(options.Namespace, viper.GetString("kubeconfig"))
+	if err != nil {
+		return errors.Wrap(err, "creating kudo client")
+	}
+
+	return update(instanceToUpdate, kc, options)
+}
+
+func update(instanceToUpdate string, kc *kudo.Client, options *updateOptions) error {
+	// Make sure the instance you want to upgrade exists
+	instance, err := kc.GetInstance(instanceToUpdate, options.Namespace)
+	if err != nil {
+		return errors.Wrapf(err, "verifying the instance does not already exist")
+	}
+	if instance == nil {
+		return fmt.Errorf("instance %s in namespace %s does not exist in the cluster", instanceToUpdate, options.Namespace)
+	}
+
+	// Change instance to point to the new OV and optionally update parameters
+	err = kc.UpdateInstance(instanceToUpdate, options.Namespace, nil, options.Parameters)
```

Comment:
also, why do we pass `nil` as `operatorVersionName` here?

Reply:
because we don't want to update the version in this call. I use nil here as I would use `None` in scala